src/Tests/Microsoft.Diagnostics.Monitoring.WebApi.UnitTests/Operation/EgressOperationStoreTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Moq;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.UnitTests.Operation
+{
+    public class EgressOperationStoreTests
+    {
+        private const string AllowOperationKey = "allowed";
+        private const string DenyOperationKey = "denied";
+
+        private static EgressOperationStore SetupStore()
+        {
+            bool allowOperations;
+
+            Mock<IRequestLimitTracker> mockRequestLimitTracker = new();
+            mockRequestLimitTracker.Setup(tracker => tracker.Increment(It.IsAny<string>(), out allowOperations))
+                .Callback((string key, out bool allowOperations) =>
+                {
+                    allowOperations = key.Equals(AllowOperationKey, StringComparison.OrdinalIgnoreCase);
+                })
+                .Returns(Mock.Of<IDisposable>());
+
+            Mock<IServiceProvider> mockServiceProvider = new();
+            mockServiceProvider.Setup(provider => provider.GetService(typeof(IRequestLimitTracker)))
+                .Returns(mockRequestLimitTracker.Object);
+
+            mockServiceProvider.Setup(provider => provider.GetService(typeof(IEgressOperationQueue)))
+                .Returns(Mock.Of<IEgressOperationQueue>());
+
+            return new EgressOperationStore(mockServiceProvider.Object);
+        }
+
+        [Fact]
+        public async Task AddOperation_RespectsRequestLimit()
+        {
+            // Arrange
+            EgressOperationStore store = SetupStore();
+
+            // Act & Assert
+            _ = await store.AddOperation(Mock.Of<IEgressOperation>(), AllowOperationKey);
+            await Assert.ThrowsAsync<TooManyRequestsException>(() => store.AddOperation(Mock.Of<IEgressOperation>(), DenyOperationKey));
+        }
+
+        [Fact]
+        public async Task StopOperation_OnException_InvokesCallback()
+        {
+            // Arrange
+            EgressOperationStore store = SetupStore();
+
+            Mock<IEgressOperation> mockOperation = new();
```

Comment:
I'm not opposed to this, but I'm curious how it would compare with just creating test implementations of these interfaces, in terms of readability and verbosity.